src/vision/status_quo/barbara_wishes_for_easy_runtime_switch.md

```diff
@@ -0,0 +1,59 @@
+# 😱 Status quo stories: Barbara wishes for an easy runtime switch
+
+## 🚧 Warning: Draft status 🚧
+
+This is a draft "status quo" story submitted as part of the brainstorming period. It is derived from
+real-life experiences of actual Rust users and is meant to reflect some of the challenges that Async
+Rust programmers face today.
+
+## The story
+
+[Barbara] has been working on an async codebase for the past 5 years. It is extremely mature and
+quite large (in the millions of lines of code). The product has traditionally been run on the server
+only, but now they want to move large parts to the edge. After some investigation, they realize that
+the open-source async runtime they have been using (runtime X), while great for server workloads, is
+not very performant on the low-end edge devices.
+
+Runtime X makes tradeoffs in memory for performance and while performance is still a concern on
+the low-end edge devices Barbara is targetting, memory consumption is even a hard constraint there.
+Additionally, the usage of a threadpool by runtime X does not help either as the targetted edge
+devices have either a single CPU thread or only a few.
+
+They had never planned to run on edge devices and so it was assumed that they would always stick
+to runtime X. Barbara looks into the how much work it would be to convert their codebase from
+runtime X to a custom runtime tuned to their workload. To their disapointment, they discover many
+areas, where their choice of runtime permeates the code base:
+
+* They extensively use runtime-dependent helpers (e.g., a select! macro).
+* The timers, tasks, and async locks implementation all assume runtime X.
+* Above all else, runtime X uses a custom version of `AsyncRead` and `AsyncWrite` traits which
+  differ from the ones used by rest of the ecosystem.
+
+They investigate the possibility of providing some sort of compatibility layer between runtime X
+and their new runtime of choice but this turns out to not seem like the right way to go as this
+compatibility layer would require too much overhead.
+
+Barbara makes the decision to rewrite large parts of the codebase to use the new runtime. This is
+extremely expensive and takes much longer than she originally thought. Barbara is very disapointed.
+
+## 🤔 Frequently Asked Questions
+
+*Here are some standard FAQ to get you started. Feel free to add more!*
+
+### **What are the morals of the story?**
+* Using a certain executor often means using a certain run-time ecosystem. This often locks the user into that ecosystem.
+* Tying yourself to a certain executor means that you are tied to the priorities of that executor.
+    * You may be happy with the run-time ecosystem, but have special needs that the default executor does not provide. If the executor doesn't have an extensibility model, you're stuck.
```

Comment:
I would add the point that it's reasonable that general purpose executors won't always be willing or able to handle specialized needs.